[Writing Tools] Rename Writing Tools methods

rr-codes opened the pull request:
#### 88e0d4623c8d3afce5dd71538244a7e2c1fe1cf8
<pre>
[Writing Tools] Rename Writing Tools methods
<a href="https://bugs.webkit.org/show_bug.cgi?id=275670">https://bugs.webkit.org/show_bug.cgi?id=275670</a>
<a href="https://rdar.apple.com/130162952">rdar://130162952</a>

Reviewed by Abrar Rahman Protyasha.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/loader/EmptyClients.cpp:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebCore::ChromeClient::proofreadingSessionUpdateStateForSuggestionWithID):
(WebCore::ChromeClient::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect): Deleted.
(WebCore::ChromeClient::textReplacementSessionUpdateStateForReplacementWithID): Deleted.
* Source/WebCore/page/ContextMenuClient.h:
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::contextMenuItemSelected):
(WebCore::ContextMenuController::populate):
(WebCore::ContextMenuController::checkOrEnableIfNeeded const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::willBeginWritingToolsSession):
(WebCore::Page::didBeginWritingToolsSession):
(WebCore::Page::proofreadingSessionDidReceiveSuggestions):
(WebCore::Page::proofreadingSessionDidUpdateStateForSuggestion):
(WebCore::Page::didEndWritingToolsSession):
(WebCore::Page::compositionSessionDidReceiveTextWithReplacementRange):
(WebCore::Page::updateStateForSelectedSuggestionIfNeeded):
(WebCore::Page::writingToolsSessionDidReceiveAction):
(WebCore::Page::willBeginTextReplacementSession): Deleted.
(WebCore::Page::didBeginTextReplacementSession): Deleted.
(WebCore::Page::textReplacementSessionDidReceiveReplacements): Deleted.
(WebCore::Page::textReplacementSessionDidUpdateStateForReplacement): Deleted.
(WebCore::Page::didEndTextReplacementSession): Deleted.
(WebCore::Page::textReplacementSessionDidReceiveTextWithReplacementRange): Deleted.
(WebCore::Page::updateStateForSelectedReplacementIfNeeded): Deleted.
(WebCore::Page::textReplacementSessionDidReceiveEditAction): Deleted.
* Source/WebCore/page/Page.h:
* Source/WebCore/page/writing-tools/WritingToolsController.h:
* Source/WebCore/page/writing-tools/WritingToolsController.mm:
(WebCore::WritingToolsController::willBeginWritingToolsSession):
(WebCore::WritingToolsController::didBeginWritingToolsSession):
(WebCore::WritingToolsController::proofreadingSessionDidReceiveSuggestions):
(WebCore::WritingToolsController::proofreadingSessionDidUpdateStateForSuggestion):
(WebCore::WritingToolsController::compositionSessionDidReceiveTextWithReplacementRange):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction&lt;WritingTools::Session::Type::Proofreading&gt;):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction&lt;WritingTools::Session::Type::Composition&gt;):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction):
(WebCore::WritingToolsController::didEndWritingToolsSession&lt;WritingTools::Session::Type::Proofreading&gt;):
(WebCore::WritingToolsController::didEndWritingToolsSession&lt;WritingTools::Session::Type::Composition&gt;):
(WebCore::WritingToolsController::didEndWritingToolsSession):
(WebCore::WritingToolsController::updateStateForSelectedSuggestionIfNeeded):
(WebCore::WritingToolsController::willBeginTextReplacementSession): Deleted.
(WebCore::WritingToolsController::didBeginTextReplacementSession): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidReceiveReplacements): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidUpdateStateForReplacement): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidReceiveTextWithReplacementRange): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidReceiveEditAction&lt;WritingTools::Session::Type::Proofreading&gt;): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidReceiveEditAction&lt;WritingTools::Session::Type::Composition&gt;): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidReceiveEditAction): Deleted.
(WebCore::WritingToolsController::didEndTextReplacementSession&lt;WritingTools::Session::Type::Proofreading&gt;): Deleted.
(WebCore::WritingToolsController::didEndTextReplacementSession&lt;WritingTools::Session::Type::Composition&gt;): Deleted.
(WebCore::WritingToolsController::didEndTextReplacementSession): Deleted.
(WebCore::WritingToolsController::updateStateForSelectedReplacementIfNeeded): Deleted.
* Source/WebCore/platform/ContextMenuItem.cpp:
(WebCore::isValidContextMenuAction):
* Source/WebCore/platform/ContextMenuItem.h:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::contextMenuItemTagWritingTools):
(WebCore::contextMenuItemTagSwapCharacters): Deleted.
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebCore/platform/Logging.h:
* Source/WebKit/Shared/API/c/WKContextMenuItemTypes.h:
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
(WebKit::toAPI):
(WebKit::toImpl):
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::writingToolsBehavior const):
(API::PageConfiguration::setWritingToolsBehavior):
(API::PageConfiguration::unifiedTextReplacementBehavior const): Deleted.
(API::PageConfiguration::setUnifiedTextReplacementBehavior): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKMenuItemIdentifiers.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKMenuItemIdentifiersPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _initializeWithConfiguration:]):
(-[WKWebView isWritingToolsActive]):
(-[WKWebView willBeginWritingToolsSession:requestContexts:]):
(-[WKWebView didBeginWritingToolsSession:contexts:]):
(-[WKWebView proofreadingSession:didReceiveSuggestions:processedRange:inContext:finished:]):
(-[WKWebView proofreadingSession:didUpdateState:forSuggestionWithUUID:inContext:]):
(-[WKWebView didEndWritingToolsSession:accepted:]):
(-[WKWebView compositionSession:didReceiveText:replacementRange:inContext:finished:]):
(-[WKWebView writingToolsSession:didReceiveAction:]):
(-[WKWebView _proofreadingSessionWithUUID:showDetailsForSuggestionWithUUID:relativeToRect:]):
(-[WKWebView _proofreadingSessionWithUUID:updateState:forSuggestionWithUUID:]):
(-[WKWebView _textReplacementSession:showInformationForReplacementWithUUID:relativeToRect:]): Deleted.
(-[WKWebView _textReplacementSession:updateState:forReplacementWithUUID:]): Deleted.
(-[WKWebView _isUnifiedTextReplacementActive]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration setWritingToolsBehavior:]):
(-[WKWebViewConfiguration writingToolsBehavior]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfigurationPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
(-[WKWebView _web_wantsWritingToolsInlineEditing]):
(-[WKWebView _web_wantsCompleteUnifiedTextReplacementBehavior]): Deleted.
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::PageClientImplCocoa::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::PageClientImplCocoa::writingToolsActiveWillChange):
(WebKit::PageClientImplCocoa::writingToolsActiveDidChange):
(WebKit::PageClientImplCocoa::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect): Deleted.
(WebKit::PageClientImplCocoa::textReplacementSessionUpdateStateForReplacementWithID): Deleted.
(WebKit::PageClientImplCocoa::unifiedTextReplacementActiveWillChange): Deleted.
(WebKit::PageClientImplCocoa::unifiedTextReplacementActiveDidChange): Deleted.
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::canHandleContextMenuWritingTools const):
(WebKit::WebPageProxy::handleContextMenuWritingTools):
(WebKit::WebPageProxy::setWritingToolsActive):
(WebKit::WebPageProxy::willBeginWritingToolsSession):
(WebKit::WebPageProxy::didBeginWritingToolsSession):
(WebKit::WebPageProxy::proofreadingSessionDidReceiveSuggestions):
(WebKit::WebPageProxy::proofreadingSessionDidUpdateStateForSuggestion):
(WebKit::WebPageProxy::didEndWritingToolsSession):
(WebKit::WebPageProxy::compositionSessionDidReceiveTextWithReplacementRange):
(WebKit::WebPageProxy::writingToolsSessionDidReceiveAction):
(WebKit::WebPageProxy::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebPageProxy::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebPageProxy::canHandleSwapCharacters const): Deleted.
(WebKit::WebPageProxy::handleContextMenuSwapCharacters): Deleted.
(WebKit::WebPageProxy::setUnifiedTextReplacementActive): Deleted.
(WebKit::WebPageProxy::willBeginTextReplacementSession): Deleted.
(WebKit::WebPageProxy::didBeginTextReplacementSession): Deleted.
(WebKit::WebPageProxy::textReplacementSessionDidReceiveReplacements): Deleted.
(WebKit::WebPageProxy::textReplacementSessionDidUpdateStateForReplacement): Deleted.
(WebKit::WebPageProxy::didEndTextReplacementSession): Deleted.
(WebKit::WebPageProxy::textReplacementSessionDidReceiveTextWithReplacementRange): Deleted.
(WebKit::WebPageProxy::textReplacementSessionDidReceiveEditAction): Deleted.
(WebKit::WebPageProxy::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect): Deleted.
(WebKit::WebPageProxy::textReplacementSessionUpdateStateForReplacementWithID): Deleted.
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView canPerformActionForWebView:withSender:]):
(-[WKContentView _updateTextInputTraits:]):
(-[WKContentView unifiedTextReplacementBehavior]): Deleted.
(-[WKContentView _updateTextInputTraitsForUnifiedTextReplacement:]): Deleted.
* Source/WebKit/UIProcess/ios/WKExtendedTextInputTraits.mm:
(-[WKExtendedTextInputTraits restoreDefaultValues]):
(-[WKExtendedTextInputTraits restoreDefaultWritingToolsBehaviorValue]):
(-[WKExtendedTextInputTraits restoreDefaultUnifiedTextReplacementBehaviorValue]): Deleted.
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::canHandleContextMenuWritingTools const):
(WebKit::PageClientImpl::handleContextMenuWritingTools):
(WebKit::PageClientImpl::canHandleSwapCharacters const): Deleted.
(WebKit::PageClientImpl::handleContextMenuSwapCharacters): Deleted.
* Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.mm:
(WebKit::menuItemIdentifier):
(WebKit::WebContextMenuProxyMac::getContextMenuFromItems):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::selectionDidChange):
(WebKit::WebViewImpl::canHandleContextMenuWritingTools const):
(WebKit::WebViewImpl::handleContextMenuWritingTools):
(WebKit::WebViewImpl::unifiedTextReplacementBehavior const): Deleted.
(WebKit::WebViewImpl::wantsCompleteUnifiedTextReplacementBehavior const): Deleted.
(WebKit::WebViewImpl::canHandleSwapCharacters const): Deleted.
(WebKit::WebViewImpl::handleContextMenuSwapCharacters): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebChromeClient::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebChromeClient::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect): Deleted.
(WebKit::WebChromeClient::textReplacementSessionUpdateStateForReplacementWithID): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebContextMenuClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebContextMenuClientMac.mm:
(WebKit::WebContextMenuClient::handleWritingTools):
(WebKit::WebContextMenuClient::handleSwapCharacters): Deleted.
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::willBeginWritingToolsSession):
(WebKit::WebPage::didBeginWritingToolsSession):
(WebKit::WebPage::proofreadingSessionDidReceiveSuggestions):
(WebKit::WebPage::proofreadingSessionDidUpdateStateForSuggestion):
(WebKit::WebPage::didEndWritingToolsSession):
(WebKit::WebPage::compositionSessionDidReceiveTextWithReplacementRange):
(WebKit::WebPage::writingToolsSessionDidReceiveAction):
(WebKit::WebPage::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebPage::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebPage::willBeginTextReplacementSession): Deleted.
(WebKit::WebPage::didBeginTextReplacementSession): Deleted.
(WebKit::WebPage::textReplacementSessionDidReceiveReplacements): Deleted.
(WebKit::WebPage::textReplacementSessionDidUpdateStateForReplacement): Deleted.
(WebKit::WebPage::didEndTextReplacementSession): Deleted.
(WebKit::WebPage::textReplacementSessionDidReceiveTextWithReplacementRange): Deleted.
(WebKit::WebPage::textReplacementSessionDidReceiveEditAction): Deleted.
(WebKit::WebPage::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect): Deleted.
(WebKit::WebPage::textReplacementSessionUpdateStateForReplacementWithID): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didChangeSelection):
(WebKit::WebPage::handleContextMenuWritingTools):
(WebKit::WebPage::handleContextMenuSwapCharacters): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKitLegacy/mac/WebCoreSupport/WebContextMenuClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebContextMenuClient.mm:
(WebContextMenuClient::handleWritingTools):
(WebContextMenuClient::handleSwapCharacters): Deleted.
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(toAction):
(toTag):
* Source/WebKitLegacy/mac/WebView/WebUIDelegatePrivate.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm:
(TEST(WritingTools, APIWithBehaviorNone)):
(TEST(WritingTools, APIWithBehaviorDefault)):
(TEST(WritingTools, APIWithBehaviorComplete)):

Canonical link: <a href="https://commits.webkit.org/280205@main">https://commits.webkit.org/280205@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/983b3f68ef4509ba43877addb80d15ee1306e084

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55941 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35265 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8409 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58940 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6373 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58067 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42886 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6569 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45047 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4405 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57970 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33158 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48235 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26185 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29941 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5563 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4516 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51909 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5834 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60527 "Built successfully") | 
| | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-1-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5954 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52476 "Passed tests") | 
| | [⏳ 🛠 vision-sim ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48304 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51995 "Passed tests") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8280 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31094 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32178 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33260 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31926 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebCore/page/writing-tools/WritingToolsController.mm:
```diff
@@ -212,33 +212,33 @@
+void WritingToolsController::proofreadingSessionDidUpdateStateForSuggestion(const WritingTools::Session& session, WritingTools::TextSuggestion::State newTextSuggestionState, const WritingTools::TextSuggestion& textSuggestion, const WritingTools::Context&)
 {
-    RELEASE_LOG(UnifiedTextReplacement, "WritingToolsController::textReplacementSessionDidUpdateStateForReplacement (%s) [new state: %hhu, replacement: %s]", session.identifier.toString().utf8().data(), enumToUnderlyingType(newTextSuggestionState), textSuggestion.identifier.toString().utf8().data());
+    RELEASE_LOG(WritingTools, "WritingToolsController::proofreadingSessionDidUpdateStateForSuggestion (%s) [new state: %hhu, replacement: %s]", session.identifier.toString().utf8().data(), enumToUnderlyingType(newTextSuggestionState), textSuggestion.identifier.toString().utf8().data());
```
```suggestion
    RELEASE_LOG(WritingTools, "WritingToolsController::proofreadingSessionDidUpdateStateForSuggestion (%s) [new state: %hhu, suggestion: %s]", session.identifier.toString().utf8().data(), enumToUnderlyingType(newTextSuggestionState), textSuggestion.identifier.toString().utf8().data());
```

rr-codes committed:
[Writing Tools] Rename Writing Tools methods
https://bugs.webkit.org/show_bug.cgi?id=275670
rdar://130162952

Reviewed by Abrar Rahman Protyasha.

* Source/WebCore/en.lproj/Localizable.strings:
* Source/WebCore/loader/EmptyClients.cpp:
* Source/WebCore/page/ChromeClient.h:
(WebCore::ChromeClient::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebCore::ChromeClient::proofreadingSessionUpdateStateForSuggestionWithID):
(WebCore::ChromeClient::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect): Deleted.
(WebCore::ChromeClient::textReplacementSessionUpdateStateForReplacementWithID): Deleted.
* Source/WebCore/page/ContextMenuClient.h:
* Source/WebCore/page/ContextMenuController.cpp:
(WebCore::ContextMenuController::contextMenuItemSelected):
(WebCore::ContextMenuController::populate):
(WebCore::ContextMenuController::checkOrEnableIfNeeded const):
* Source/WebCore/page/Page.cpp:
(WebCore::Page::willBeginWritingToolsSession):
(WebCore::Page::didBeginWritingToolsSession):
(WebCore::Page::proofreadingSessionDidReceiveSuggestions):
(WebCore::Page::proofreadingSessionDidUpdateStateForSuggestion):
(WebCore::Page::didEndWritingToolsSession):
(WebCore::Page::compositionSessionDidReceiveTextWithReplacementRange):
(WebCore::Page::updateStateForSelectedSuggestionIfNeeded):
(WebCore::Page::writingToolsSessionDidReceiveAction):
(WebCore::Page::willBeginTextReplacementSession): Deleted.
(WebCore::Page::didBeginTextReplacementSession): Deleted.
(WebCore::Page::textReplacementSessionDidReceiveReplacements): Deleted.
(WebCore::Page::textReplacementSessionDidUpdateStateForReplacement): Deleted.
(WebCore::Page::didEndTextReplacementSession): Deleted.
(WebCore::Page::textReplacementSessionDidReceiveTextWithReplacementRange): Deleted.
(WebCore::Page::updateStateForSelectedReplacementIfNeeded): Deleted.
(WebCore::Page::textReplacementSessionDidReceiveEditAction): Deleted.
* Source/WebCore/page/Page.h:
* Source/WebCore/page/writing-tools/WritingToolsController.h:
* Source/WebCore/page/writing-tools/WritingToolsController.mm:
(WebCore::WritingToolsController::willBeginWritingToolsSession):
(WebCore::WritingToolsController::didBeginWritingToolsSession):
(WebCore::WritingToolsController::proofreadingSessionDidReceiveSuggestions):
(WebCore::WritingToolsController::proofreadingSessionDidUpdateStateForSuggestion):
(WebCore::WritingToolsController::compositionSessionDidReceiveTextWithReplacementRange):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction<WritingTools::Session::Type::Proofreading>):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction<WritingTools::Session::Type::Composition>):
(WebCore::WritingToolsController::writingToolsSessionDidReceiveAction):
(WebCore::WritingToolsController::didEndWritingToolsSession<WritingTools::Session::Type::Proofreading>):
(WebCore::WritingToolsController::didEndWritingToolsSession<WritingTools::Session::Type::Composition>):
(WebCore::WritingToolsController::didEndWritingToolsSession):
(WebCore::WritingToolsController::updateStateForSelectedSuggestionIfNeeded):
(WebCore::WritingToolsController::willBeginTextReplacementSession): Deleted.
(WebCore::WritingToolsController::didBeginTextReplacementSession): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidReceiveReplacements): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidUpdateStateForReplacement): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidReceiveTextWithReplacementRange): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidReceiveEditAction<WritingTools::Session::Type::Proofreading>): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidReceiveEditAction<WritingTools::Session::Type::Composition>): Deleted.
(WebCore::WritingToolsController::textReplacementSessionDidReceiveEditAction): Deleted.
(WebCore::WritingToolsController::didEndTextReplacementSession<WritingTools::Session::Type::Proofreading>): Deleted.
(WebCore::WritingToolsController::didEndTextReplacementSession<WritingTools::Session::Type::Composition>): Deleted.
(WebCore::WritingToolsController::didEndTextReplacementSession): Deleted.
(WebCore::WritingToolsController::updateStateForSelectedReplacementIfNeeded): Deleted.
* Source/WebCore/platform/ContextMenuItem.cpp:
(WebCore::isValidContextMenuAction):
* Source/WebCore/platform/ContextMenuItem.h:
* Source/WebCore/platform/LocalizedStrings.cpp:
(WebCore::contextMenuItemTagWritingTools):
(WebCore::contextMenuItemTagSwapCharacters): Deleted.
* Source/WebCore/platform/LocalizedStrings.h:
* Source/WebCore/platform/Logging.h:
* Source/WebKit/Shared/API/c/WKContextMenuItemTypes.h:
* Source/WebKit/Shared/API/c/WKSharedAPICast.h:
(WebKit::toAPI):
(WebKit::toImpl):
* Source/WebKit/UIProcess/API/APIPageConfiguration.h:
(API::PageConfiguration::writingToolsBehavior const):
(API::PageConfiguration::setWritingToolsBehavior):
(API::PageConfiguration::unifiedTextReplacementBehavior const): Deleted.
(API::PageConfiguration::setUnifiedTextReplacementBehavior): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKMenuItemIdentifiers.mm:
* Source/WebKit/UIProcess/API/Cocoa/WKMenuItemIdentifiersPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _initializeWithConfiguration:]):
(-[WKWebView isWritingToolsActive]):
(-[WKWebView willBeginWritingToolsSession:requestContexts:]):
(-[WKWebView didBeginWritingToolsSession:contexts:]):
(-[WKWebView proofreadingSession:didReceiveSuggestions:processedRange:inContext:finished:]):
(-[WKWebView proofreadingSession:didUpdateState:forSuggestionWithUUID:inContext:]):
(-[WKWebView didEndWritingToolsSession:accepted:]):
(-[WKWebView compositionSession:didReceiveText:replacementRange:inContext:finished:]):
(-[WKWebView writingToolsSession:didReceiveAction:]):
(-[WKWebView _proofreadingSessionWithUUID:showDetailsForSuggestionWithUUID:relativeToRect:]):
(-[WKWebView _proofreadingSessionWithUUID:updateState:forSuggestionWithUUID:]):
(-[WKWebView _textReplacementSession:showInformationForReplacementWithUUID:relativeToRect:]): Deleted.
(-[WKWebView _textReplacementSession:updateState:forReplacementWithUUID:]): Deleted.
(-[WKWebView _isUnifiedTextReplacementActive]): Deleted.
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration setWritingToolsBehavior:]):
(-[WKWebViewConfiguration writingToolsBehavior]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfigurationPrivate.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewInternal.h:
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Source/WebKit/UIProcess/API/mac/WKWebViewMac.mm:
(-[WKWebView _web_wantsWritingToolsInlineEditing]):
(-[WKWebView _web_wantsCompleteUnifiedTextReplacementBehavior]): Deleted.
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.h:
* Source/WebKit/UIProcess/Cocoa/PageClientImplCocoa.mm:
(WebKit::PageClientImplCocoa::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::PageClientImplCocoa::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::PageClientImplCocoa::writingToolsActiveWillChange):
(WebKit::PageClientImplCocoa::writingToolsActiveDidChange):
(WebKit::PageClientImplCocoa::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect): Deleted.
(WebKit::PageClientImplCocoa::textReplacementSessionUpdateStateForReplacementWithID): Deleted.
(WebKit::PageClientImplCocoa::unifiedTextReplacementActiveWillChange): Deleted.
(WebKit::PageClientImplCocoa::unifiedTextReplacementActiveDidChange): Deleted.
* Source/WebKit/UIProcess/Cocoa/WebPageProxyCocoa.mm:
(WebKit::WebPageProxy::canHandleContextMenuWritingTools const):
(WebKit::WebPageProxy::handleContextMenuWritingTools):
(WebKit::WebPageProxy::setWritingToolsActive):
(WebKit::WebPageProxy::willBeginWritingToolsSession):
(WebKit::WebPageProxy::didBeginWritingToolsSession):
(WebKit::WebPageProxy::proofreadingSessionDidReceiveSuggestions):
(WebKit::WebPageProxy::proofreadingSessionDidUpdateStateForSuggestion):
(WebKit::WebPageProxy::didEndWritingToolsSession):
(WebKit::WebPageProxy::compositionSessionDidReceiveTextWithReplacementRange):
(WebKit::WebPageProxy::writingToolsSessionDidReceiveAction):
(WebKit::WebPageProxy::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebPageProxy::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebPageProxy::canHandleSwapCharacters const): Deleted.
(WebKit::WebPageProxy::handleContextMenuSwapCharacters): Deleted.
(WebKit::WebPageProxy::setUnifiedTextReplacementActive): Deleted.
(WebKit::WebPageProxy::willBeginTextReplacementSession): Deleted.
(WebKit::WebPageProxy::didBeginTextReplacementSession): Deleted.
(WebKit::WebPageProxy::textReplacementSessionDidReceiveReplacements): Deleted.
(WebKit::WebPageProxy::textReplacementSessionDidUpdateStateForReplacement): Deleted.
(WebKit::WebPageProxy::didEndTextReplacementSession): Deleted.
(WebKit::WebPageProxy::textReplacementSessionDidReceiveTextWithReplacementRange): Deleted.
(WebKit::WebPageProxy::textReplacementSessionDidReceiveEditAction): Deleted.
(WebKit::WebPageProxy::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect): Deleted.
(WebKit::WebPageProxy::textReplacementSessionUpdateStateForReplacementWithID): Deleted.
* Source/WebKit/UIProcess/PageClient.h:
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView canPerformActionForWebView:withSender:]):
(-[WKContentView _updateTextInputTraits:]):
(-[WKContentView unifiedTextReplacementBehavior]): Deleted.
(-[WKContentView _updateTextInputTraitsForUnifiedTextReplacement:]): Deleted.
* Source/WebKit/UIProcess/ios/WKExtendedTextInputTraits.mm:
(-[WKExtendedTextInputTraits restoreDefaultValues]):
(-[WKExtendedTextInputTraits restoreDefaultWritingToolsBehaviorValue]):
(-[WKExtendedTextInputTraits restoreDefaultUnifiedTextReplacementBehaviorValue]): Deleted.
* Source/WebKit/UIProcess/mac/PageClientImplMac.h:
* Source/WebKit/UIProcess/mac/PageClientImplMac.mm:
(WebKit::PageClientImpl::canHandleContextMenuWritingTools const):
(WebKit::PageClientImpl::handleContextMenuWritingTools):
(WebKit::PageClientImpl::canHandleSwapCharacters const): Deleted.
(WebKit::PageClientImpl::handleContextMenuSwapCharacters): Deleted.
* Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.mm:
(WebKit::menuItemIdentifier):
(WebKit::WebContextMenuProxyMac::getContextMenuFromItems):
* Source/WebKit/UIProcess/mac/WebViewImpl.h:
* Source/WebKit/UIProcess/mac/WebViewImpl.mm:
(WebKit::WebViewImpl::selectionDidChange):
(WebKit::WebViewImpl::canHandleContextMenuWritingTools const):
(WebKit::WebViewImpl::handleContextMenuWritingTools):
(WebKit::WebViewImpl::unifiedTextReplacementBehavior const): Deleted.
(WebKit::WebViewImpl::wantsCompleteUnifiedTextReplacementBehavior const): Deleted.
(WebKit::WebViewImpl::canHandleSwapCharacters const): Deleted.
(WebKit::WebViewImpl::handleContextMenuSwapCharacters): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebChromeClient::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebChromeClient::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect): Deleted.
(WebKit::WebChromeClient::textReplacementSessionUpdateStateForReplacementWithID): Deleted.
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/WebContextMenuClient.h:
* Source/WebKit/WebProcess/WebCoreSupport/mac/WebContextMenuClientMac.mm:
(WebKit::WebContextMenuClient::handleWritingTools):
(WebKit::WebContextMenuClient::handleSwapCharacters): Deleted.
* Source/WebKit/WebProcess/WebPage/Cocoa/WebPageCocoa.mm:
(WebKit::WebPage::willBeginWritingToolsSession):
(WebKit::WebPage::didBeginWritingToolsSession):
(WebKit::WebPage::proofreadingSessionDidReceiveSuggestions):
(WebKit::WebPage::proofreadingSessionDidUpdateStateForSuggestion):
(WebKit::WebPage::didEndWritingToolsSession):
(WebKit::WebPage::compositionSessionDidReceiveTextWithReplacementRange):
(WebKit::WebPage::writingToolsSessionDidReceiveAction):
(WebKit::WebPage::proofreadingSessionShowDetailsForSuggestionWithIDRelativeToRect):
(WebKit::WebPage::proofreadingSessionUpdateStateForSuggestionWithID):
(WebKit::WebPage::willBeginTextReplacementSession): Deleted.
(WebKit::WebPage::didBeginTextReplacementSession): Deleted.
(WebKit::WebPage::textReplacementSessionDidReceiveReplacements): Deleted.
(WebKit::WebPage::textReplacementSessionDidUpdateStateForReplacement): Deleted.
(WebKit::WebPage::didEndTextReplacementSession): Deleted.
(WebKit::WebPage::textReplacementSessionDidReceiveTextWithReplacementRange): Deleted.
(WebKit::WebPage::textReplacementSessionDidReceiveEditAction): Deleted.
(WebKit::WebPage::textReplacementSessionShowInformationForReplacementWithIDRelativeToRect): Deleted.
(WebKit::WebPage::textReplacementSessionUpdateStateForReplacementWithID): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::didChangeSelection):
(WebKit::WebPage::handleContextMenuWritingTools):
(WebKit::WebPage::handleContextMenuSwapCharacters): Deleted.
* Source/WebKit/WebProcess/WebPage/WebPage.h:
* Source/WebKit/WebProcess/WebPage/WebPage.messages.in:
* Source/WebKitLegacy/mac/WebCoreSupport/WebContextMenuClient.h:
* Source/WebKitLegacy/mac/WebCoreSupport/WebContextMenuClient.mm:
(WebContextMenuClient::handleWritingTools):
(WebContextMenuClient::handleSwapCharacters): Deleted.
* Source/WebKitLegacy/mac/WebView/WebHTMLView.mm:
(toAction):
(toTag):
* Source/WebKitLegacy/mac/WebView/WebUIDelegatePrivate.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WritingTools.mm:
(TEST(WritingTools, APIWithBehaviorNone)):
(TEST(WritingTools, APIWithBehaviorDefault)):
(TEST(WritingTools, APIWithBehaviorComplete)):

Canonical link: https://commits.webkit.org/280205@main
